tsl/src/nodes/decompress_chunk/compressed_batch.c

```diff
@@ -225,6 +225,36 @@ static void
 compute_plain_qual(DecompressContext *dcontext, DecompressBatchState *batch_state, Node *qual,
 				   uint64 *restrict result)
 {
+	/*
+	 * Some predicates can be evaluated to a Const at run time.
+	 */
+	if (IsA(qual, Const))
+	{
+		Const *c = castNode(Const, qual);
+		if (c->constisnull || !DatumGetBool(c->constvalue))
+		{
+			/*
+			 * Some predicates are evaluated to a null Const, like a
+			 * strict comparison with stable expression that evaluates to null.
+			 * No rows pass.
+			 */
+			const size_t n_batch_result_words = (batch_state->total_batch_rows + 63) / 64;
+			for (size_t i = 0; i < n_batch_result_words; i++)
+			{
+				result[i] = 0;
+			}
+		}
+		else
+		{
+			/*
+			 * This is a constant true qual, every row passes and we can
+			 * just ignore it. No idea how it can happen though.
+			 */
+			Assert(false);
```

Comment:
I wonder if we should log an error here to ensure we also catch these cases in production builds.

Reply:
The good thing is that this case doesn't require any handling, so I'd just leave it as is, the query will work correctly even if a user manages to hit it somehow.